File: src/main/java/com/wgzhao/addax/admin/service/AddaxLogService.java
package com.wgzhao.addax.admin.service;

import com.wgzhao.addax.admin.dto.AddaxLogDto;
import com.wgzhao.addax.admin.model.AddaxLog;
import com.wgzhao.addax.admin.model.EtlStatistic;
import com.wgzhao.addax.admin.model.TbAddaxSta;
import com.wgzhao.addax.admin.repository.AddaxLogRepo;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;
import java.util.Optional;

@Service
public class AddaxLogService
{
    @Autowired
    private AddaxLogRepo addaxLogRepo;

    public void insertLog(long tid,  String message) {
        AddaxLog addaxLog = new AddaxLog();
        addaxLog.setTid(tid);
        addaxLog.setLog(message);
        addaxLog.setRunDate(LocalDate.now());
        addaxLog.setRunAt(LocalDateTime.now());
        addaxLogRepo.save(addaxLog);
    }

    public List<LocalDate> getLast5RunDatesByTid(long tid) {
        return addaxLogRepo.findTop5ByTidOrderByRunDateDesc(tid).stream()
                .map(AddaxLog::getRunDate)
                .distinct()
                .toList();
    }
    public AddaxLog getLastLogByTid(long tid) {
        return addaxLogRepo.findFirstByTidOrderByRunDateDesc(tid);
    }

    public List<AddaxLog> getLast5LogsById(long tid) {
        return addaxLogRepo.findTop5ByTidOrderByRunDateDesc(tid);
    }

    public String getLogContent(long id) {
        return addaxLogRepo.findLogById(id);
    }

    public List<AddaxLogDto> getLogEntry(String tid)
    {
        return addaxLogRepo.findLogEntry(tid);
    }

}

File: src/main/kotlin/com/wgzhao/addax/admin/service/AddaxLogService.kt
package com.wgzhao.addax.admin.service

import com.wgzhao.addax.admin.dto.AddaxLogDto
import com.wgzhao.addax.admin.model.AddaxLog
import com.wgzhao.addax.admin.repository.AddaxLogRepo
import io.github.oshai.kotlinlogging.KotlinLogging
import org.springframework.stereotype.Service
import java.time.LocalDate
import java.time.LocalDateTime

@Service
class AddaxLogService(
    private val addaxLogRepo: AddaxLogRepo
) {
    private val log = KotlinLogging.logger {}

    fun insertLog(tid: Long, message: String?) {
        val addaxLog = AddaxLog(tid=tid, log=message, runDate = LocalDate.now(), runAt = LocalDateTime.now() )
        addaxLogRepo.save(addaxLog)
    }

    fun getLast5RunDatesByTid(tid: Long): List<LocalDate?>? =
        addaxLogRepo.findTop5ByTidOrderByRunDateDesc(tid)?.mapNotNull { it?.runDate }

    fun getLastLogByTid(tid: Long): AddaxLog? =
        addaxLogRepo.findFirstByTidOrderByRunDateDesc(tid)

    fun getLast5LogsById(tid: Long): List<AddaxLog?>? =
        addaxLogRepo.findTop5ByTidOrderByRunDateDesc(tid)

    fun getLogContent(id: Long): String? =
        addaxLogRepo.findById(id).orElse(null)?.log

    fun getLogEntry(tid: Long?): List<AddaxLogDto> =
        addaxLogRepo.findTop5ByTidOrderByRunAtDesc(tid) ?: emptyList()
}
